lib/util.js

```diff
@@ -0,0 +1,106 @@
+const bs58 = require('bs58')
+/**
+ * @typedef {import('./range').Range} Range
+ * @typedef {import('ppppp-db/msg-v4').Msg} Msg
+ */
+
+/**
+ * @param {any} msgId
+ * @return {msgId is string}
+ */
+function isMsgId (msgId) {
+  try {
+    const d = bs58.decode(msgId)
+    return d.length === 32
+  } catch {
+    return false
+  }
+}
+
+/**
+ * @param {any} msgIds
+ * @return {msgIds is Array<string>}
+ */
+function isMsgIds(msgIds) {
+  if (!Array.isArray(msgIds)) return false
+  return msgIds.every(isMsgId)
+}
+
+/**
+ * @param {any} msgs
+ * @return {msgs is Array<Msg>}
+ */
+function isMsgs(msgs) {
+  if (!Array.isArray(msgs)) return false
+  return msgs.every(isMsg)
+}
+
+/**
+ * @param {any} range
+ * @return {range is Range}
```

Comment:
Sweet :)

Reply:
i added this because i felt fancy but it didn't seem to affect the types in the end. i have some vague memory that you need a stricter ts type mode for this to have an effect.